pf/internal/check/check_test.go

```diff
@@ -96,6 +154,80 @@ func TestSensitiveIDWithOverride(t *testing.T) {
 	})
 }
 
+func TestInvalidInputID(t *testing.T) {
+	tests := []struct {
+		name          string
+		idSchema      idSchema
+		expectedError bool
+	}{
+		{name: "Required", idSchema: idSchema{required: true, optional: false, computed: false}, expectedError: true},
+		{name: "Optional", idSchema: idSchema{required: false, optional: true, computed: false}, expectedError: true},
+		{name: "Computed", idSchema: idSchema{required: false, optional: false, computed: true}, expectedError: false},
+		{name: "Optional+Computed", idSchema: idSchema{required: false, optional: true, computed: true}, expectedError: true},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name+"true", func(t *testing.T) {
+			stderr, err := test(t, tfbridge.ProviderInfo{
+				P: pfbridge.ShimProvider(testProvider{withID: &idSchema{required: true}}),
+			})
+			assert.NotEmpty(t, stderr)
+			assert.Error(t, err)
+		})
+		t.Run(tc.name+"true (missing ComputeID)", func(t *testing.T) {
+			stderr, err := test(t, tfbridge.ProviderInfo{
+				P: pfbridge.ShimProvider(testProvider{withID: &tc.idSchema}),
+				Resources: map[string]*tfbridge.ResourceInfo{
+					"test_res": {Fields: map[string]*tfbridge.SchemaInfo{
+						"id": {Name: "otherId"},
+					}},
+				},
+			})
+			if tc.expectedError {
+				assert.NotEmpty(t, stderr)
+				assert.Error(t, err)
+			} else {
+				assert.Empty(t, stderr)
+				assert.NoError(t, err)
+			}
+		})
+		t.Run(tc.name+"true (missing Name)", func(t *testing.T) {
+			stderr, err := test(t, tfbridge.ProviderInfo{
+				P: pfbridge.ShimProvider(testProvider{withID: &tc.idSchema}),
+				Resources: map[string]*tfbridge.ResourceInfo{
+					"test_res": {Fields: map[string]*tfbridge.SchemaInfo{
+						"id": {},
+					}, ComputeID: func(ctx context.Context, state property.PropertyMap) (property.ID, error) {
+						panic("ComputeID")
+					}},
+				},
+			})
+			if tc.expectedError {
+				assert.NotEmpty(t, stderr)
+				assert.Error(t, err)
+			} else {
+				assert.Empty(t, stderr)
+				assert.NoError(t, err)
```

Comment:
Curious if we need to permit this to not error out, it seems weird to allow ComputeID when there is a usable "id" field, but maybe that's ok, that could mean that the author wants ComputeID to override what is used for identity, even if there is a natural one.  If we think it's ok, should this  error in other cases?  Perhaps if ComputeID take care of identity then an "id" field could be just a normal field and we don't care if it's in/out?

Reply:
The original issue this is trying to solve is that schema gen will not generate an "id" input property so if there is one then you _must_ rename it to something else. And if you rename it to something else, then you _must_ provide `ComputeID`.

And yes it is weird to provide a ComputeID when there is a valid "id", but since it is allowed and you could do it I think we have to allow it.